daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -841,11 +841,69 @@ func (devices *DeviceSet) createRegisterDevice(hash string) (*devInfo, error) {
 	return info, nil
 }
 
-func (devices *DeviceSet) createRegisterSnapDevice(hash string, baseInfo *devInfo, size uint64) error {
-	if err := devices.poolHasFreeSpace(); err != nil {
+func (devices *DeviceSet) takeSnapshot(hash string, baseInfo *devInfo, size uint64) error {
+	var (
+		doSuspend      bool
```

Comment:
We need to declare this (doSuspend), as the first time it is used, It is scoped inside an if condition.